syntax = "proto3";

package dwarf;

// Kamera öffnen
message ReqOpenCamera {
  bool binning = 1;           // Binning aktivieren
  int32 rtsp_encode_type = 2; // 0: 4K, 1: 1080p
}

// Foto aufnehmen
message ReqPhoto {
}

// Serienaufnahme starten
message ReqBurst {
  int32 count = 1;  // Anzahl Aufnahmen
}

// Video starten
message ReqStartRecord {
}

// Video stoppen
message ReqStopRecord {
}

// Alle Parameter setzen
message ReqSetAllParams {
  int32 exp_mode = 1;        // 0: Auto, 1: Manuell
  int32 exp_index = 2;       // Belichtungs-Index
  int32 gain_mode = 3;       // Verstärkungs-Modus
  int32 gain_index = 4;      // Verstärkungs-Index
  int32 ircut_value = 5;     // 0: CUT, 1: PASS
  int32 wb_mode = 6;         // 0: Auto, 1: Manuell
  int32 wb_index_type = 7;   // 0: Farbtemperatur, 1: Szene
  int32 wb_index = 8;        // Weißabgleich-Index
  int32 brightness = 9;      // Helligkeit (0-255)
  int32 contrast = 10;       // Kontrast (0-255)
  int32 hue = 11;            // Farbton (0-255)
  int32 saturation = 12;     // Sättigung (0-255)
  int32 sharpness = 13;      // Schärfe (0-100)
  int32 jpg_quality = 14;    // JPG-Qualität
}

// Alle Parameter abrufen
message ReqGetAllParams {
}

// Response mit allen Parametern
message ResGetAllParams {
  int32 exp_mode = 1;
  int32 exp_index = 2;
  int32 gain_mode = 3;
  int32 gain_index = 4;
  int32 ircut_value = 5;
  int32 wb_mode = 6;
  int32 wb_index_type = 7;
  int32 wb_index = 8;
  int32 brightness = 9;
  int32 contrast = 10;
  int32 hue = 11;
  int32 saturation = 12;
  int32 sharpness = 13;
  int32 jpg_quality = 14;
  int32 code = 15;
}

// Zeitraffer starten
message ReqStartTimelapsePhoto {
  int32 interval = 1;  // Intervall in Sekunden
  int32 count = 2;     // Anzahl Aufnahmen
}

// System-Status
message ResSystemWorkingState {
  int32 code = 1;
  int32 state = 2;  // Aktueller Status
}
